client/serviceinterface_update.go

```diff
@@ -329,7 +329,10 @@ func removeServiceInterfaceTarget(serviceName string, targetName string, deleteI
 			modified := false
 			targets := []types.ServiceInterfaceTarget{}
 			for _, t := range service.Targets {
-				if (t.Name == targetName || (t.Name == "" && targetName == serviceName)) && (t.Namespace == "" || t.Namespace == namespace) {
+				targetFound := t.Name == targetName && t.Service == ""
+				targetServiceFound := t.Service == targetName
+				namespaceMatches := t.Namespace == "" || t.Namespace == namespace
```

Comment:
I think the logic would be clearer as

`
targetFound := (t.Name == targetName && t.Service == "") && t.Namespace == namespace
targetServiceFound := t.Service == targetName && t.Namespace == ""
if targetFound || targetServiceFound) {

`

Reply:
Certainly more precise! I will update it.